tests/ui/conftest.py

```diff
@@ -0,0 +1,144 @@
+# -*- coding: utf-8 -*-
+
+"""Plugin for running UI tests."""
+
+import socket
+
+import pytest
+
+from redash import (
+    create_app,
+    redis_connection,
+)
+from redash.models import db
+from tests.factories import Factory
+from tests.ui.live_server import LiveServer
+
+
+@pytest.fixture(name='host')
+def fixture_host():
+    """Return a host name as str."""
+    return 'localhost'
+
+
+@pytest.fixture(name='port')
+def fixture_port():
+    """Return a free port as int."""
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    s.bind(('', 0))
+    port = s.getsockname()[1]
+    s.close()
+    return port
+
+
+@pytest.fixture(name='app')
+def fixture_app(host, port):
+    """Run setup and teardown code for the Flask app."""
+    app = create_app('testing')
+
+    app.config['TESTING'] = True
+    app.config['SERVER_NAME'] = '{host}:{port}'.format(host=host, port=port)
+
+    app_ctx = app.app_context()
+    app_ctx.push()
+
+    db.session.close()
+    db.drop_all()
+    db.create_all()
+
+    yield app
```

Comment:
I would try.. finally this yield so the cleanup is always done no matter what happens in the test

Reply:
pytest takes care of that

```python
import pytest

@pytest.fixture
def foobar():
    print('\n>setup')
    yield
    print('\n>teardown')

def test_foobar(foobar):
    raise RuntimeError
```

```text
$ pytest -s -v test_foobar.py
test_foobar.py::test_foobar
>setup
FAILED
>teardown
```